libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt

```diff
@@ -120,12 +126,37 @@ fun NavigationRoute.refreshRoute(
                 updateLegs = updateLegs,
                 waypoints = buildNewWaypoints(this@refreshRoute.waypoints, waypoints)
             )
+            .updateRefreshTtl(this.unrecognizedJsonProperties, refreshTtl)
             .build()
     }
     val directionsResponseBlock: DirectionsResponse.Builder.() -> DirectionsResponse.Builder = {
         waypoints(buildNewWaypoints(directionsResponse.waypoints(), waypoints))
     }
-    return update(directionsRouteBlock, directionsResponseBlock)
+    val newExpirationTime = refreshTtl?.plus(responseTime)
+    return update(directionsRouteBlock, directionsResponseBlock, newExpirationTime ?: expirationTime)
+}
+
+private fun DirectionsRoute.Builder.updateRefreshTtl(
```

Comment:
do we need to keep actual ttl value at all? Maybe expiration time will be enough?

Reply:
We don't actually use it, you are right, but I wanted consistency here. We update all other stuff we receive from refresh. So it makes sense to update refresh_ttl too. Otherwise it'll be hard to keep track of what's updated and what's not.
What if, for example, in the future we won't have this internal expirationTime property, but we'll have responseTime in the DirectionsResponse/DirectionsRefreshResponse out of the box? Then updating refresh_ttl is critical: we'll need the latest value in our calculations.
But at the moment it's mostly for consistency.